hibernate-core/src/main/java/org/hibernate/dialect/MySQLDialect.java

```diff
@@ -541,4 +536,47 @@ public IdentityColumnSupport getIdentityColumnSupport() {
 	public boolean isJdbcLogWarningsEnabledByDefault() {
 		return false;
 	}
+
+	@Override
+	public boolean supportsCascadeDelete() {
+		return getMySQLStorageEngine().supportsCascadeDelete();
+	}
+
+	@Override
+	public String getTableTypeString() {
+		return getMySQLStorageEngine().getTableTypeString(getEngineKeyword());
+	}
+
+	protected String getEngineKeyword() {
+		return "type";
+	}
+
+	@Override
+	public boolean hasSelfReferentialForeignKeyBug() {
+		return getMySQLStorageEngine().hasSelfReferentialForeignKeyBug();
+	}
+
+	@Override
+	public boolean dropConstraints() {
+		return getMySQLStorageEngine().dropConstraints();
+	}
+
+	protected MySQLStorageEngine getMySQLStorageEngine() {
+		String storageEngine = Environment.getProperties().getProperty( Environment.STORAGE_ENGINE );
+		if(storageEngine == null) {
+			storageEngine = System.getProperty( Environment.STORAGE_ENGINE );
+		}
+		if(storageEngine == null) {
+			return InnoDBStorageEngine.INSTANCE;
```

Comment:
I created a new getDefaultMySQLStorageEngine method:

	protected MySQLStorageEngine getDefaultMySQLStorageEngine() {
		return MyISAMStorageEngine.INSTANCE;
	}

which is overridden in MySQL55Dialect and set to InnoDB.